### DISCUSSION TITLE
8.0.0 Released

### DISCUSSION THREAD

**Discussion by adrianhall** [CommunityToolkit/Datasync#79] [Announcements]
Hey folks,

Release 8.0.0 - the first release of the Datasync Community Toolkit - has just been released to NuGet.  Since it's the first release, expect it to take a while to turn up (I'll be checking it tomorrow AM).  I'd consider this **BETA** quality.

In addition, we have documentation at https://CommunityToolkit.github.io/Datasync - this is one area that everyone can easily contribute to:

* If you find an error in the docs, file an issue - include the page and what the problem is.
* If you want to fix an error, fork the repository, fix the problem, and send me a PR.

Finally, we have new samples.  Currently, there is a WinUI3 sample client and an ASP.NET Core server for it.  However, I'll be expanding those over the coming weeks.

Also to do (and it's a little higher priority for me) - automated testing of the client against real servers.

Once again - this is your call to try out the library, feed me problems, ideas, and contribute to the documentation and samples.

**Comment by henda79** [comment]
Hey Adrian, the link to the docs above returns a 404. Still no sign of the NuGet... I checked 11 hours after your post and included prerelease too.

**Reply by hansmbakker** [reply]
the url is https://communitytoolkit.github.io/Datasync, there is a typo above (missing n)

**Reply by henda79** [reply]
ah yes, thanks.

**Comment by adrianhall** [comment]
Just to update everyone (where are the packages?)  We are having some teething problems with the first release, and need to get some approvals done so that we can publish under the guise of the dotNET Foundation.  Hopefully, these will be resolved soon.

**Reply by henda79** [reply]
ok, I'm eager to quickly implement this so I'll keep a look out for it.

I've had a quick flick through the docs. Is the server side predominately the same ?

**Reply by adrianhall** [reply]
Yes - just namespace changes, really, although there are a few changes in EntityFrameworkCore when you are using PostgreSQL, MongoDB, or Cosmos.

**Reply by myokeeh** [reply]
@adrianhall per the docs link above, it's `dotnet new -i Microsoft.AspNetCore.Datasync.Template.CSharp` and not `dotnet new -i CommunityToolkit.Datasync.Server.Template.CSharp`? Is the former temporary or is it going to be that moving forward?

**Reply by adrianhall** [reply]
See issue #106 - I'm having problems with the package signing, which I need to fix.  Once that is done, it will be CommunityToolkit.Datasync.Server.Template.CSharp - the other one is from Azure Mobile Apps and not the same thing.

**Comment by adrianhall** [comment]
Two things have come up while I have been doing the integration tests:

1) Serious - if you do two PullAsync() operations on the same instance of the context, the query description is modified to adjust for the delta token, but it is stored as the default query description, thus modifying all future pullasync() methods.  Solution: only do one PullAsync() per context.

2) Less serious (but still serious) - Entity Framework Core is not thread-safe. When you do PushAsync() with multiple threads, the context can get confused.  The practical upshot of this is that entities are not removed from the operations queue due to conflicting changes to the context. 

Fortunately, I know the solution to both of these and they will be done by the end of the day.  However, 8.0.0 is not recommended right now.
